src/biogeochem/CNPhenologyMod.F90

```diff
@@ -1776,36 +1773,32 @@ subroutine CropPhenology(num_pcropp, filter_pcropp                     , &
          ! from AgroIBIS subroutine planting
          ! ---------------------------------
 
-         ! in order to allow a crop to be planted only once each year
-         ! initialize cropplant = .false., but hold it = .true. through the end of the year
-
          ! initialize other variables that are calculated for crops
          ! on an annual basis in cropresidue subroutine
 
-         if ( jday == jdayyrstart(h) .and. mcsec == 0 )then
-
-            ! make sure variables aren't changed at beginning of the year
-            ! for a crop that is currently planted, such as
-            ! WINTER TEMPERATE CEREAL = winter (wheat + barley + rye)
-            ! represented here by the winter wheat pft
-
-            if (.not. croplive(p))  then
-               cropplant(p) = .false.
-               idop(p)      = NOT_Planted
-
-               ! keep next for continuous, annual winter temperate cereal crop;
-               ! if we removed elseif,
-               ! winter cereal grown continuously would amount to a cereal/fallow
-               ! rotation because cereal would only be planted every other year
-
-            else if (croplive(p) .and. (ivt(p) == nwwheat .or. ivt(p) == nirrig_wwheat)) then
-               cropplant(p) = .false.
-               !           else ! not possible to have croplive and ivt==cornORsoy? (slevis)
-            end if
+         ! Second condition ensures everything is correctly set when resuming from a run with old code
+         ! OR starting a run mid-year without any restart file OR handling a new crop column that just
+         ! came into existence (and not at the year boundary for some reason).
+         if ( is_beg_curr_year() .or. crop_inst%sdates_thisyr(p,1) == spval ) then
```

Comment:
I actually think there is another possible subtle issue here: CropPhenology is currently only called when doalb is .true. Based on a test I just ran, in an I compset, doalb is .false. for both time steps 0 and 1, so CropPhenology is first called in the second time step of Jan 1 – meaning this conditional wouldn't be triggered in the first year of the run. If we're starting from cold start or an old restart file, then the spval check here should handle that situation – doing this start-of-year resetting the first time CropPhenology is called, even though `is_beg_curr_year` will be false. However, I think this will currently cause problems if you start a new run from a restart file created from this branch (not via a real restart, i.e., `CONTINUE_RUN`, but a new startup case with finidat set to point to a restart file created from this branch). If I'm thinking about things right, in this situation, `sdates_thisyr` will typically *not* be `spval` (at least, not after you add it to the restart file, which I think is needed for other scenarios) but `is_beg_curr_year` won't be triggered in the first year either – and so `sowing_count`, `harvest_count`, etc. won't get reset in the first year in this scenario.

I think the easiest way to fix this is to get rid of the `doalb` check around the call to `CropPhenology`, but I want to get some confirmation from others that that is an okay thing to change. @samsrabin please comment if you have any thoughts on this.

- [ ] I am putting a place-holder checkbox here to remind us that this will need to be resolved in some way. I'm hoping that we can fix this in the course of addressing #1623, probably by changing the conditional around the CropPhenology call to no longer check `doalb`. But if not, I think we'll need to come up with a different fix for this issue.

Reply:
Yes, I think that logic works.